src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10600,13 +10600,31 @@ private ImplicitArrayCreationExpressionSyntax ParseImplicitlyTypedArrayCreation(
             var commas = _pool.Allocate();
             try
             {
-                while (this.CurrentToken.Kind == SyntaxKind.CommaToken)
+                while (true)
                 {
-                    commas.Add(this.EatToken());
+                    if (this.IsPossibleExpression())
+                    {
+                        var size = this.AddError(this.ParseExpressionCore(), ErrorCode.ERR_InvalidArray);
```

Comment:
💯  If you don't mind I incorporate this part in the stackalloc inits PR

Reply:
i would mind that greatly.  roslyn is built on non-cooperation, and it should stay that way.